### PR TITLE
Feature/remove checks on max requester secrets count

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java [hunk 1]
```diff
@@ -264,7 +263,10 @@ void shouldGetPalaemonTokensWithoutAppComputeSecret() {
                         IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "2", "file2"));
         assertThat(tokens)
                 .containsEntry(IEXEC_APP_DEVELOPER_SECRET_0, "");
-        assertThat(tokens.get(REQUESTER_SECRETS)).isEqualTo(Collections.emptyMap());
+        assertThat(tokens.get(REQUESTER_SECRETS)).isEqualTo(Map.of(
```
Comment (zguesmi) [inline src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java:266]: Why isn't it emplty anymore?

File: src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java [hunk 2]
```diff
@@ -252,20 +249,11 @@ private Map<String, Object> getApplicationComputeSecrets(TaskDescription taskDes
         tokens.put(IexecEnvUtils.IEXEC_APP_DEVELOPER_SECRET_PREFIX + secretIndex, appDeveloperSecret0);
 
         final HashMap<String, String> requesterSecrets = new HashMap<>();
-        Optional<TeeTaskComputeSecretCount> oMaxApplicationSecretIndex = teeTaskComputeSecretCountService.getMaxAppComputeSecretCount(applicationAddress, SecretOwnerRole.REQUESTER);
```
Comment (zguesmi) [inline src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java:255]: Maybe we should add `@Deprecated` or a comment to the method teeTaskComputeSecretCountService.getMaxAppComputeSecretCount()` to explain why it is not used and that is intended to be remove in the future.

File: src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java (hunk 1 above)
Comment (jbern0rd) [reply src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java:266]: As the test is designed, requester secrets in deal parameters are not empty.
Previously, as the max requester secret count was not defined, there was a barrier and map was empty.

Now and following yesterday discussion, the deal parameter should be considered consistent and as being validated previously in the process.

Maybe the test can be split in 2 tests:
1. If deal parameters have secrets which does not exist, it produces the map with empty values (see https://github.com/iExecBlockchainComputing/iexec-sms/blob/feature/remove-checks-on-max-requester-secrets-count/src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java#L272)
2. if deal parameters do not have secrets, the map should be empty (we need a test without calling this method https://github.com/iExecBlockchainComputing/iexec-sms/blob/feature/remove-checks-on-max-requester-secrets-count/src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java#L409)

What do you think ?

File: src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java (hunk 2 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java:255]: I will see how do to it cleanly.
Globally, it should be all the `TeeTaskComputeSecretCountService` which becomes deprecated and could/should be deprecated.

File: src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java (hunk 1 above)
Comment (zguesmi) [reply src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java:266]: Thanks for the explanation.
I do agree on what you suggest, splitting the test seems good to me.

Comment (jbern0rd) [reply src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java]: Tests reworked a little bit in 0c43867

File: src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java (hunk 2 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionService.java:286]: Everything has been removed

**Review by mcornaton — APPROVED** [review]
Thanks for cleaning!

**Review by zguesmi — APPROVED** [review]
:100:

**Pull request merged**